app/models/miq_queue.rb

```diff
@@ -153,6 +153,19 @@ def self.put(options)
     msg
   end
 
+  # Execute a job on all servers.  Filterable by zone
+  #
+  def self.broadcast(options)
+    server_targets = MiqServer.select(:id, :guid).select("zones.name AS zone_name")
+                              .joins(:zone)
+
+    server_targets = server_targets.where(:zone_id => options[:zone].id) if options[:zone]
+
+    server_targets.each do |server|
+      put(options.merge(:server_guid => server.guid, :zone => server["zone_name"]))
```

Comment:
You should be able to do server.zone_name

Reply:
Ah, missed the options hash usage here. So the `:zone` key is used for filtering and also as a queue option? And then we also expect other options in the hash to pass to the actual queue message? That's a confusing API. I would rather:

```ruby
def self.broadcast(zone, queue_options)
...
end
```

After all the zone in the queue message isn't needed if we're always passing the guid.

Reply:
Additionally we're expecting a zone object here where the other queue messages expect the zone name. That inconsistency makes the option hash even more difficult to understand.

Reply:
:+1: for the zone object/name consistency with other methods like put.

Reply:
> Additionally we're expecting a zone object here where the other queue messages expect the zone name.

For the record, I was confused that `MiqServer.my_zone` (used by `Zone.determine_queue_zone(options)`) returned a `zone.name` and was under the impression what `.determine_queue_zone` was returning an object as well, not a string.  Would have passed in a string for `:zone` had I known that.

In my opinion, the MiqServer public API is a confusing API:

```ruby
MiqServer.my_server
#=> <#MiqServer ...>
MiqServer.my_zone
#=> "zone name" (String)
```

Regardless, I have deleted this, but this is a note that the intent was to match the `MiqQueue.put` argument structure.